packages/relay-compiler/core/RelaySchemaUtils.js

```diff
@@ -214,6 +214,14 @@ function assertTypeWithFields(type: GraphQLType): GraphQLObjectType | GraphQLInt
   return type;
 }
 
+// TODO: Remove this when flow types for `getNullableType` in graphql-js
```

Comment:
@leebyron I tried fixing it with the current type signature but could not figure out the issue. Here's the flow error I get:

```
packages/relay-compiler/handlers/connection/RelayConnectionTransform.js:352
352:   const a = RelaySchemaUtils.getNullableType(type);
                                                  ^^^^ GraphQLEnumType. This typ
e is incompatible with
 30:   GraphQLScalarType |
       ^^^^^^^^^^^^^^^^^ GraphQLScalarType. See: node_modules/graphql/type/defin
ition.js.flow:30

packages/relay-compiler/handlers/connection/RelayConnectionTransform.js:352
352:   const a = RelaySchemaUtils.getNullableType(type);
                                                  ^^^^ GraphQLInputObjectType. T
his type is incompatible with
 30:   GraphQLScalarType |
       ^^^^^^^^^^^^^^^^^ GraphQLScalarType. See: node_modules/graphql/type/defin
ition.js.flow:30

packages/relay-compiler/handlers/connection/RelayConnectionTransform.js:352
352:   const a = RelaySchemaUtils.getNullableType(type);
                                                  ^^^^ GraphQLInterfaceType. Thi
s type is incompatible with
 30:   GraphQLScalarType |
       ^^^^^^^^^^^^^^^^^ GraphQLScalarType. See: node_modules/graphql/type/defin
ition.js.flow:30

packages/relay-compiler/handlers/connection/RelayConnectionTransform.js:352
352:   const a = RelaySchemaUtils.getNullableType(type);
                                                  ^^^^ GraphQLList. This type is
 incompatible with
 30:   GraphQLScalarType |
       ^^^^^^^^^^^^^^^^^ GraphQLScalarType. See: node_modules/graphql/type/defin
ition.js.flow:30

packages/relay-compiler/handlers/connection/RelayConnectionTransform.js:352
352:   const a = RelaySchemaUtils.getNullableType(type);
                                                  ^^^^ GraphQLNonNull. This type
 is incompatible with
 30:   GraphQLScalarType |
       ^^^^^^^^^^^^^^^^^ GraphQLScalarType. See: node_modules/graphql/type/defin
ition.js.flow:30

packages/relay-compiler/handlers/connection/RelayConnectionTransform.js:352
352:   const a = RelaySchemaUtils.getNullableType(type);
                                                  ^^^^ GraphQLObjectType. This t
ype is incompatible with
 30:   GraphQLScalarType |
       ^^^^^^^^^^^^^^^^^ GraphQLScalarType. See: node_modules/graphql/type/defin
ition.js.flow:30

packages/relay-compiler/handlers/connection/RelayConnectionTransform.js:352
352:   const a = RelaySchemaUtils.getNullableType(type);
                                                  ^^^^ GraphQLUnionType. This ty
pe is incompatible with
 30:   GraphQLScalarType |
       ^^^^^^^^^^^^^^^^^ GraphQLScalarType. See: node_modules/graphql/type/defin
ition.js.flow:30


Found 7 errors
```

Kind of confused about what is causing it since `type` is of type `GraphQLType` so I don't see where the issue could be. Maybe flow is having trouble typechecking this?

AFAIK the new type signature should not make it miss potential errors, it will just lose type refinements. IE passing it `GraphQLScalarType | GraphQLNonNull<*>` would return `GraphQLNullableType` which is a wider union instead of `GraphQLScalarType` with the current version.